test/test-utils/src/main/java/software/amazon/awssdk/testutils/EnvironmentVariableHelper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.testutils;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.rules.ExternalResource;
+
+/**
+ * A utility that can temporarily forcibly set environment variables and
+ * then allows resetting them to the original values.
+ */
+public class EnvironmentVariableHelper extends ExternalResource {
```

Comment:
I've seen one of these elsewhere as well. Was this just a rename?

Reply:
Where? I think this is the first time I've seen it

Reply:
It looks like it was just a method in the profile test that I remembered: https://github.com/aws/aws-sdk-java-v2/blob/master/core/src/test/java/software/amazon/awssdk/auth/ProfileCredentialsProviderTest.java#L231-L240. We're good.

Reply:
I'll see if it can be replaced with this utility.